### PR TITLE
[compiled autograd] clear compiled_autograd_verbose once test is done

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #126483
* __->__ #126148

verbose flag leaks into tests ran after

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_compiled_autograd.py [hunk 1]
```diff
@@ -1534,6 +1534,7 @@ def fn():
         self.assertEqual(
             sum(1 for e in expected_logs if e in logs.getvalue()), len(expected_logs)
         )
+        torch._logging.set_logs(compiled_autograd_verbose=False)
```
Comment (peterbell10) [inline test/inductor/test_compiled_autograd.py:1537]: These should also be in try-finally blocks so we restore the state even if the test fails.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**